src/shared/libcrypt-util.c

```diff
@@ -75,21 +75,23 @@ int make_salt(char **ret) {
 #endif
 }
 
-int hash_password(const char *password, char **ret) {
+int hash_password_full(const char *password, void **cd_data, int *cd_size, char **ret) {
```

Comment:
i still wouldn't expose the cd_data, sounds like needless optimization. i.e it's the kind of optimization you might want to do ina hash breaker or so, but otherwise I really wouldn't bother...

Reply:
This is exposed here, but is only used in the test. I think it is worthwhile to show that in tests: it should make it much easier to debug issues like #16965.